apps/dashboard/app/models/script.rb

```diff
@@ -152,6 +152,7 @@ def destroy
   end
 
   def update(params)
+    params ||= {}
```

Comment:
Not sure why this is here - Rails should always populate params. In any case I think `params` is a `HashWithIndifferentAccess` which could be important in case we're using strings and symbols here and there.

Reply:
When all form fields are removed from the edit script page and then the form is submitted, the application will throw a null pointer and show the stacktrace on the screen. When there are no fields on the form, params is `nil` in update.

Reply:
```ruby
# ScriptsController#51
@script.update(save_script_params[:script])
```
`save_script_params[:script] `will return nil when the form is empty

Reply:
> When all form fields are removed from the edit script page and then the form is submitted

I haven't looked this over in detail quite yet - but you shouldn't be able to remove _all_ fields. You need _at least_ `auto_script`.

Reply:
OK - at the moment you can. I will revert this and take a look to disable deleting `auto_scripts` from the form.

Reply:
Fixed